feat(provider/docker): Add metrics to docker registry provider to instrument number of calls made (#2963)

Remz-Jay opened the pull request:
See https://github.com/spinnaker/spinnaker/issues/2963

Specs are missing; will be added once approach is approved.

Remz-Jay committed:
feat(provider/docker): Add metrics to docker registry provider to instrument number of calls made (#2963)

wheleph commented on clouddriver-docker/src/main/groovy/com/netflix/spinnaker/clouddriver/docker/registry/metrics/UrlMetricsInstrumentation.groovy:
```diff
@@ -0,0 +1,52 @@
+    registry.timer(timingId.withTag('url', response.request().url().toString())).record(end - start, TimeUnit.MILLISECONDS)
+    String status = (response.successful) ? 'success' : 'failure'
+    registry.counter(counterId.withTag('url', response.request().url().toString()).withTag('status', status)).increment()
+    log.info format("Received response for %s in %.1fms", response.request().url(), (end - start) / 1e6d)
```
Personally for me `1_000_000.0` is more readable than `1e6d`

pvdissel replied:
Which is the minimal supported Java version for Spinnaker/Clouddriver? If 7, then `1_000_000.0` is superb, if older than 7, than it's not supported (https://docs.oracle.com/javase/8/docs/technotes/guides/language/underscores-literals.html)

pvdissel replied:
Better yet, turn `1e6d` or `1_000_000.0` into a static variable with an useful name, so we actually know what this magic number means

mdirkse replied:
Better yet, use the JDK time methods to do the transform so that it's clear what you're doing: https://docs.oracle.com/javase/8/docs/api/java/time/package-summary.html

Remz-Jay replied:
Updated, using java.time. Start and End are measured using `Instant`, and then a `Duration` is calculated between them which is then converted to milliseconds. No more `1e6d` magic!

Remz-Jay commented on clouddriver-docker/src/main/groovy/com/netflix/spinnaker/clouddriver/docker/registry/metrics/UrlMetricsInstrumentation.groovy:
```diff
@@ -0,0 +1,52 @@
+  }
+
+  void interceptorCompleted(Response response, long start, long end) {
+    registry.timer(timingId.withTag('url', response.request().url().toString())).record(end - start, TimeUnit.MILLISECONDS)
```
As @mdirkse said: We should split `url` into `host` and `path`, and put them in separate tags. That way we can also do aggregations top level registry for instance.

Remz-Jay replied:
Updated. Metrics are now recorded with a `host` and `path` tag, instead of one `url` tag. Usability is limited in our own scenario, since `host` is always the same.

pvdissel reviewed (approved):
LGTM

Remz-Jay committed:
refactor(provider/docker): Fix copyright and url metrics namespace + casing (#2963)

- Most metrics under Clouddriver are namespaced (like redis.command.*, jedis.pool.*). Applied the same for url metrics; namespaced under httpClient.url.*.
- Updated the className shown for url metrics from UrlMetricsInstrumentation to DockerRegistryClient, which actually owns the httpClient that emits these metrics. Should make for more logical labeling.

mdirkse committed:
fix(provider/kubernetes): upgrade to latest version of spinnaker-dependencies to fix spinnaker/spinnaker#3082 (#2807)

posox committed:
fix(provider/openstack): fix several issues in glance v2 support (#2787)

* location value can be empty
* convert image properties to Map<String, String>

tomaslin committed:
feat(provider/aws): r5d instance type (#2813)

cfieber committed:
fix(perf): fix performance regression due to default provider cache metrics causing extra redis reads. (#2816)

Remz-Jay committed:
refactor(provider/docker): Prevent unbounded {path} metrics (#2963)

- Adds Docker Registry specific interpretation to the URLs being instrumented;
  They are now split into "tags/list", tagged with the repository + image being requested,
  and "other", which contains all other calls. Further distinction for "other" URLs can
  be made using the supplied log line.
- Add statusCode, containing 2xx, 4xx, 5xx, etc.
- Renamed "host" to "registry" since it matches the registry:repository:image stanza better.
- Removed the "httpClient.url.count" counter, since "httpClient.url.requestTime" already
  contains a counter which exactly the same data.
- Refactored the test to reflect these changes and made it compatible with upstream changes.

Commit:
feat(provider/kubernetes): v2 Add support for api service kind (#2798)

gojenkuser committed:
feat(provider/kubernetes): Adds support for strategy.spinnaker.io/use… (#2812)

* feat(provider/kubernetes): Adds support for strategy.spinnaker.io/use SourceCapacity annotation.

Adds support for strategy.spinnaker.io/useSourceCapacity annotation in manifests of type CanScale to use the deployed manifest replicas value. Allows teams to scale resources without having the cluster inadvertently scale during the next deployment.

* fix build errors

cfieber committed:
fix(perf): reverts DefaultProviderCache metric changes (#2821)

Revert "fix(perf): fix performance regression due to default provider cache metrics causing extra redis reads. (#2816)"
This reverts commit 45642482c98d670df82b9853c63a9222f5aa5a65.

Revert "feat(cache): metrics on putCacheResult. (#2790)"
This reverts commit e78a1aebc28d3afadf358e7e69ed739c299e88dd.

ethanfrogers committed:
fix(artifacts): fixes multiple http base providers (#2820)

when enabling multiple http based artifact providers (http + gitlab +
github) clouddriver filed to start stating that multiple beans were
found.

Commit:
feat(provider/dcos): accommodates changes to secrets in metronome.

Commit:
feat(builds): Introduces ability to only build specific cloud provider platforms. (#2819)

Some providers are more granular than others. This would allow a developer/release engineer to specify `./gradlew -PincludePlatforms=GCE,AWS,KUBERNETES build` with any permutation of platform dependencies. However, this doesn't quite work yet, due to entanglement of the clouddriver-security project and clouddriver-web previously being the centerpoint of how all providers come together

Commit:
fix(dependencies): Remove AWS and Google modules from the clouddriver-security dependency list. (#2830)

cfieber committed:
chore(dependencies): spinnaker-dependencies 1.0.5 (#2831)

jkschneider committed:
refactor(provider/cf): delete existing cloud foundry implementation (#2828)

jeyrschabu committed:
feat(aws): Adding a delete ami operation (#2822)

- added operation for deleting an AMI in account/region

ezimanyi committed:
fix(google): Guard against null backend in disabled load balancer check (#2827)

* fix(google): Don't cache invalid TCP or Internal load balancers

If we can't find the backend for a  TCP or Internal load balancer,
we should add it to the list of failed load balancers so that it
we don't include an incomplete load balancer in the cache.

* fix(google): Guard against null backend in disabled load balancer check

* fix(google): Lower log level of warning in HTTP Load Balancer caching agent

wjoel committed:
fix(cats): Remove unsupported annotations (#2832)

Newer versions of spinnaker-dependencies do not include FindBugs.

clouddriver fails to build since upgrading spinnaker-dependencies
to 0.158.0: https://github.com/spinnaker/clouddriver/pull/2732

https://github.com/spinnaker/clouddriver/pull/2825 fixed this for
the 1.8 branch, and the discussion suggests dropping the annotations,
so that's what this commit does.

lwander committed:
feat(provider/kubernetes): check kind availability before caching (#2833)

anshrma committed:
fix(provider/aws) : Adding name tag to newly created Security Group (#2823)

posox committed:
fix(provider/kubernetes): set cluster type in k8s v2 provider (#2788)

Return cluster type "kubernetes" instead null

asher committed:
fix(cats): saner redis defaults (#2836)

emjburns committed:
fix(titus): actually ignore NOT_FOUND (#2839)

edwinavalos committed:
feat(provider/docker): adding passwordCommand option for retrieving docker credentials (#2837)

Allows users to use passwordCommand instead of password or passwordFile if their setup isn't conducive to the latter two options. This refreshes the credentials anytime the docker creds are used (every time it caches images), so an agent wasn't needed to re-initialize credentials constantly.

lwander committed:
fix(provider/kubernetes): be permissive of unreachable clusters on startup (#2840)

lwander committed:
fix(provider/kubernetes): avoid caching empty artifacts (#2841)

This is a fairly significant CPU improvement when caching large
clusters. We're seeing ~5-10% reduction in CPU usage during cache data
construction by not creating & passing around empty artifacts for every
resource deployed to a given cluster.

ajordens committed:
fix(entitytags): Bit of logging around the `UpsertEntityTagsAtomicOperation` (#2845)

ajordens committed:
fix(entitytags): Bit of logging around the `DeleteEntityTagsAtomicOperation` (#2846)

abcsnowwolf committed:
fix(provider/kuberneteres): Speed up caching server group (#2834)

caching pods for server group one time and handle them locally.

lwander committed:
fix(provider/kubernetes): deployment w/ 0 replicas is 'stable' (#2848)

lwander committed:
fix(provider/kubernetes): record if a crd is namespace-scoped (#2849)

This was generating a lot of incorrect log warnings about kinds missing
namespaces during caching

asher committed:
fix(web): fix getServerGroupsForIds method (#2853)

ajordens committed:
feat(entitytags): Reconcile elastic search if entity tags don't exist (#2855)

Might as well delete any entity tags from elastic search if we determine
they don't exist in front50.

jeyrschabu committed:
fix(aws): Adding missing `@Component` to `deregister image` converter (#2856)

- added missing `@Component`

lwander committed:
fix(provider/kubernetes): only load relevant on demand keys (#2854)

This greatly speeds up (testings shows reduction of ~25% time spent) caching
runs for large agents.

lwander committed:
fix(provider/kubernetes): don't record namespace relationship in cache (#2859)

This was generating very large (1 per resource) numbers of duplicate
cache entries that both ate up memory, and required expensive
relationship merging & deduping later. We don't actually make use of
this relationship anywhere, and it can be infered from a resource's key
in the cache, making it redundant.

lwander committed:
fix(provider/kubernetes): speed up stratifying cache elements (#2857)

Cuts down cache time by ~1% -- not a huge improvement but was discovered
making other improvements

lwander committed:
fix(provider/kubernetes): only record app relationships for clusters (#2862)

Another effort at reducing the number of cache objects being deduped and
stored in memory during cache cycles.

lwander committed:
fix(provider/kubernetes): reduce allocations during cache merge (#2863)

lwander committed:
fix(provider/kubernetes): skip redundant check of kinds already omitted (#2866)

ezimanyi committed:
fix(provider/kubernetes): Services have a cluster relationship (#2865)

anshrma committed:
feat(provider/aws) : adding useragent suffix

asher committed:
fix(cats/search): fix NPE in findMatches (#2870)

Commit:
feat(artifacts): expose types supported by a given artifact credential (#2871)

jkschneider committed:
feat(provider/cf): add provider skeleton with credentials management (#2838)

lwander committed:
fix(provider/kubernetes): make request timeout configurable (#2869)

Prevents caching agents from stacking up while failing to ever complete
when indexing unavailable clusters. Will need to be set by a
user/operator.

robzienert committed:
feat(provider/ecs): Server group events endpoint (#2877)

* First go at the ECS server group controller

* feat(provider/ecs): Added the ECS server group events endpoint

* Renamed the StatusJudge into

ServerGroupEventStatusConverter

* Updated PR as per robert zienert's comments

lwander committed:
feat(provider/kubernetes): add server group managers to sg endpoint (#2878)

jkschneider committed:
feat(provider/cf): add model and cache keys (#2875)

Co-Authored-By: Jammy Louie <jlouie@pivotal.io>
Co-Authored-By: Joris Melchior <joris.melchior@gmail.com>
Co-Authored-By: Stu Pollock <spollock@pivotal.io>

lwander committed:
feat(core): give clouddriver the option to shutdown gracefully (#2882)

* feat(core): give clouddriver the option to shutdown gracefully

* feat(core): associate tasks with owner

ezimanyi committed:
fix(core): Fix dependency in Dockerfile (#2883)

lwander committed:
fix(provider/kubernetes): reduce crd lookups when none exist (#2886)

If the call fails, the memoizer can't record the success time, voiding
the use of an expiring memoizer. When a lot of deployments happen at
once, we check all our caching agent's kinds very frequently generating
a lot of calls to the "crd" kind, which are now all memoized.

zthroo committed:
update null check to be explicit

Remz-Jay committed:
Merge branch 'master' into feature/2963

yejingtao committed:
fix(openstack): error exception message while deleting server group #2885 (#2885)

merge remote master

brantburnett committed:
fix(provider/docker): Clear docker token cache after 401 (#2818)

If a 401 is received from the Docker registry, the cached token is
cleared so it isn't reused if a new token isn't acquired. Subsequent
requests will then get a new token, if necessary.

fixes spinnaker/spinnaker#2039

lwander committed:
fix(provider/kubernetes): deployment sequence only applies to rs (#2892)

lwander committed:
chore(provider/kubernetes): helpful startup & runtime logging (#2891)

tomaslin committed:
feat(titus): adding timestamp to deploy operation log (#2895)

lwander committed:
feat(provider/kubernetes): cache pod metrics (#2896)

These aren't surfaced yet

lwander committed:
feat(provider/kubernetes): surface metrics (#2898)

Commit:
feat(titus): Add console log endpoint (#2899)

lwander committed:
fix(provider/kubernetes): security groups are related to apps (#2893)

bruno-carrier-lookout committed:
feat(provider/ecs): Support for Elastic Network Interfaces (AWSVPC networking mode) (#2852)

* feat(provider/ecs): Support for Elastic Network Interfaces (AWSVPC networking mode)

* PR comments addressed

clareliguori committed:
fix(provider/ecs): Handle null startedAt field for tasks (#2904)

lwander committed:
fix(core): use snakeyaml's SafeConstructor (#2906)

dibyom committed:
refactor(build): Move top level provider config to com.spinnaker.config (#2905)

Any Spring Configuration under com.spinnaker.config will be scanned by
`clouddriver-web` removing the necessity to specify them directly using
an `Import`.

dibyom committed:
fix(build): Add a NoopSearchableProvider (#2902)

This allows clouddriver-web to boot up without any cloudproviders.

clareliguori committed:
fix(provider/ecs): Resolve ECS security group names (#2903)

* fix(provider/ecs): Resolve security group names when creating an ECS service

* Add license headers

ezimanyi committed:
chore(build): Bump gradle plugin version

ajordens committed:
feat(providers/amazon): block device mappings for t3 (#2910)

ezimanyi committed:
fix(entitytags): Don't set _type field on Elasticsearch documents (#2913)

Storing multiple mapping types in the same index is deprecated in
Elasticsearch 6 and will be removed in Elasticsearch 7. Add a config
parameter indicating that only one mapping type per index is supported
(defaulting to false, for backwards compatibility). If that parameter
is set, all documents will have their type set to the config parameter
mappingTypeName, which defaults to Elasticsearch's recommended
value of _doc.

lwander committed:
fix(core): fix error when no matching converter is found (#2915)

The inclusion of version is a red-herring here, as this error means no
account was registered to handle the operation for the given provider

joonlim committed:
feat(deploy/ha): Add halconfig yamls for clouddriver-ro, clouddriver-rw, and clouddriver-caching. (#2914)

* feat(deploy/ha): Add halconfig yamls for clouddriver-ro, clouddriver-rw, and clouddriver-caching.

* feat(deploy/ha): Remove shared properties.

bruno-carrier-lookout committed:
feat(provider/ecs): Added support for ENIs in the instance details view (#2881)

* feat(provider/ecs): Added support for ENIs in the instance details view

* Updated tests

clareliguori committed:
feat(provider/ecs): Add server group setting for public IP address (#2908)

clareliguori committed:
feat(provider/ecs): Add support for Fargate launch type (#2909)

dibyom committed:
refactor(web): clouddriver-web should not depend on cloud provider models. (#2918)

dibyom committed:
refactor(build): Depend on aws-sdk-s3 instead of the entire sdk (#2922)

ajordens committed:
fix(aws): avoid terminating instances when lifecycle hooks are present (#2919)

This PR introduces an extension point allowing for custom behavior to
happen during a resize asg operation.

The default behavior has been modified such that instances will not be
explicitly terminated during a resize _if_ a terminating lifecycle hook
is present on the asg.

replaces spinnaker/clouddriver#2900
fixes spinnaker/spinnaker#3229

dibyom committed:
chore(dependencies): Bump spinnaker dependencies to 1.0.9 (#2924)

clareliguori committed:
fix(provider/ecs): Gracefully ignore zone lookup for tasks without a container instance (e.g. Fargate) (#2920)

lwander committed:
fix(provider/google): batch & service path must match (#2925)

shihchang committed:
fix(provider/oracle): upgrade to latest oci-sdk to fix many issues (#2874)

* fix(provider/oracle): upgrade to latest oci-sdk to fix many issues

* fix(provider/oracle): add timeCreated field to image listing response

ajordens committed:
chore(dependencies): Bump spinnaker dependencies to 1.0.10 (#2929)

ajordens committed:
fix(aws): Handle null desired capacity in `DefaultAfterResizeEventHandler` (#2932)

clareliguori committed:
fix(provider/ecs): Display VPC & security groups based on network config (#2921)

* fix(provider/ecs): Display VPC and security groups based on service network configuration if available

* Refactor tests

joonlim committed:
fix(ha): Fix clouddriver-ro/rw/caching service names to be camel-cased rather than hyphenated. (#2933)

posox committed:
fix(provider/openstack): Use directUrl property for image location (#2912)

It's a workaround for issue in Openstack4j library, locations cannot be
used without fix https://github.com/ContainX/openstack4j/pull/1096

Spinnaker issue: https://github.com/spinnaker/spinnaker/issues/3018

tomaslin committed:
Merge branch 'master' into feature/2963

ezimanyi committed:
fix(provider/google): Fix caching of >500 forwarding rules (#2934)

clareliguori committed:
fix(provider/ecs): Explicitly declare dependency for ECS account mapper on ECS creds being already initialized (#2930)

cfieber committed:
fix(projects): remove parallel cluster lookups (#2938)

A project with a large number of applications can significantly load
the redis cache by requesting all application clusters in parallel.

This removes the Rx code that was parallelizing the lookups in favor
of a linear lookup through all the project's applications, and
additionally uses the RequestQueue when performing the cluster lookups
in case we are under load and the requests are timing out.

ezimanyi committed:
refactor(provider/google): Move non-batch requests to PaginatedRequest (#2935)

* refactor(provider/google): Refactor caching agents to use PaginatedRequest

* refactor(provider/google): Move non-batch requests to PaginatedRequest

dibyom committed:
chore(dependencies): Bump gradle plugin to 4.3.0 (#2941)

Adds license reporting support

tomaslin committed:
feat(provider/titus): log run job requests for better traceability (#2942)

dibyom committed:
fix(aws): Move RoleController to clouddriver-aws (#2940)

jkschneider committed:
feat(provider/cf): API client and caching agent (#2939)

Co-Authored-By: Jammy Louie <jlouie@pivotal.io>
Co-Authored-By: Joris Melchior <joris.melchior@gmail.com>
Co-Authored-By: Stu Pollock <spollock@pivotal.io>

ajordens committed:
chore(dependencies): Bump spinnaker dependencies to 1.0.13 (#2945)

asher committed:
feat(aws): allow regex filtering of instance types (#2944)

ezimanyi committed:
fix(google): Fix bugs in pagination of caching (#2947)

Remz-Jay committed:
Merge branch 'master' into feature/2963

dibyom committed:
feat(build): Allow building only specified cloud providers (#2901)

This change allows a developer to specify a set of cloud providers
to build using the `includeCloudProviders` property e.g.
`./gradlew -PincludeCloudProviders=kubneretes, aws`
By default, all cloud providers are built.

BREAKING CHANGE

Cloud providers are now included as runtime dependencies for
the `clouddriver-web` project instead of compile time. As such,
`clouddriver-web` classes should not directly depend on cloud
provider projects. Instead, each provider project should have a
Spring `@Configuration` class in the `com.netflix.spinnaker.config`
package which will be scanned in by `clouddriver-web`.

tomaslin committed:
fix(provider/titus): need to retry job submits on DEADLINE_EXCEEDED (#2950)

jtk54 committed:
fix(provider/gce): set autoscaling cap before target size in deploys. (#2951)

In the reverse order, autoscaling policy params are set _after_
we use them to set the target size of the instance. This was
causing GCE clones to clone the new server group at the source
server group's size regardless of the value of 'useSourceCapacity'.

tomaslin committed:
fix(provider/titus): avoid resolving asg name and scaling policy copy / loadBalancer for jobs (#2953)

tomaslin committed:
Fix titus jobs (#2954)

* fix(provider/titus): avoid resolving asg name and scaling policy copy / loadBalancer for jobs

* fix(provider/titus): set name label in run job stage

robzienert committed:
perf(core): Moving project clusters generation to a caching agent (#2949)

robzienert committed:
fix(core): Handle null jenkins build info (#2955)

e96wic committed:
fix(provider/kubernetes): Improved log output from KubernetesOperationException (#2956)

robzienert committed:
fix(core): Make project clusters caching optional (#2958)

Turns out trying to cache all project clusters takes forever in larger deployments.
Rather than try to cache everything, this change makes caching opt-in project-by-project.

tomaslin committed:
fix(provider/titus): Add more logs on deployment timings (#2959)

jkschneider committed:
feat(provider/cf): add deploy, server group, and load balancer operations (#2957)

Co-Authored-By: Jammy Louie <jlouie@pivotal.io>
Co-Authored-By: Joris Melchior <joris.melchior@gmail.com>
Co-Authored-By: Stu Pollock <spollock@pivotal.io>

robzienert committed:
fix(core): Front50 perf; separate normalized allow list method (#2960)

BrunoCarrier committed:
feat(provider/ecs): Added LogDriver, LogOptions and DockerLabels (#2946)

* feat(provider/ecs): Added Docker labels, logDrivera and logOptions

* Fix unit tests

tomaslin committed:
fix(provider/titus): more logging around security group lookup (#2962)

jasonkshar committed:
feat(provider/dcos): update healthcheck to use AlwaysUpHealthIndicator class (#2894)

alanmquach committed:
feat(cats/search): Enable substring matching for instance search (#2923)

jkschneider committed:
style(provider/cf): fix license headers (#2963)

lcysimon committed:
feat(provider/ecs): Added support for environment variables (#2961)

tomaslin committed:
fix(provider/titus): don't load rules when resolving security group names and ids (#2964)

Commit:
feat(provider/cf): Open Service Broker model and operations

* Plus a little import polish

Co-Authored-By: Jammy Louie <jlouie@pivotal.io>
Co-Authored-By: Joris Melchior <joris.melchior@gmail.com>
Co-Authored-By: Stu Pollock <spollock@pivotal.io>

robzienert committed:
fix(project): Send lastPush in json responses (#2968)

shihchang committed:
fix(provider/oracle): remove oci-java-sdk-full-shaded from classpath (#2967)

lwander committed:
feat(provider/kubernetes): v2 admission controllers (#2969)

fcoelho committed:
fix(provider/ecs): fetch all images for ECR repos with >100 images (#2948)

ECR repositories may contain many images, but the `ListImages`
operation will retrieve at most 100 at a time [1]. For repositories with
more than 100 images this used to cause pipelines depending on that
repository to randomly fail when you start getting over the limit (as
the desired image would not be on the first page of results).

This commit builds upon the `ListImages` operation and the `nextToken`
value to fetch all images available in the repository. And only images
matching the desired tag/digest are sent to `DescribeImages`.

Some tests were slightly modified to include an empty list instead of
`null` in the `imageIds` field as:

1. that's what the `ListImages` returns in case there are no images in
   the repository
2. the code now uses the results from the `ListImages` operation to
   filter the images instead of the results of `DescribeImages` as
   previously done

[1]: https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/services/ecr/model/ListImagesRequest.html#getMaxResults--

lwander committed:
feat(provider/kubernetes): support PodPreset, PodSecurityPolicy (#2971)

lwander committed:
fix(provider/kubernetes): sane namespace lookup (#2972)

The prior approach of always consulting the service account namespace
was causing confusion, especially when no service account was in use

christopherthielen committed:
chore(*): Update PULL_REQUEST_TEMPLATE.md - put reviewers mention in (#2974)

backticks

christopherthielen committed:
Revert "chore(*): Update PULL_REQUEST_TEMPLATE.md - put reviewers mention in (#2974)" (#2975)

This reverts commit 3752a1b93f1e3981b46984c84d7f182492bd10ad.

varikin committed:
fix(google): Fix error to include missing load balancers. (#2973)

This error was not correctly showing the load balancers that couldn't be
found.

Commit:
fix(google): Stop retrying if allowable error encountered.

danielpeach committed:
fix(appengine): defer to configured project when using ADC (#2936)

lwander committed:
fix(provider/kubernetes): fix admission kind scopes (#2977)

lwander committed:
feat(proivder/kubernetes): allow custom resource scope config (#2978)

lwander committed:
fix(provider/kubernetes): don't return all related types from search (#2979)

We still need to search related types to find matching keys, but we
don't need to return them.

lwander committed:
refactor(provider/kubernetes): remove unused relationships (#2980)

ajordens committed:
fix(aws): Retries around security group tagging (#2982)

Creation of a security group is eventually consistent yet we were
immediately tagging a newly created security group.

This PR introduces retries _and_ treats a failure to tag as non-fatal.

dibyom committed:
refactor(google): Remove Task as a paramter for doRetry (#2981)

Allows use to remove the dependency on `clouddriver-core`.
This is mainly useful because Front50 also depends on
`clouddriver-google-common` and thereby indirectly on all
of clouddriver-core and every dependency it brings in.

lwander committed:
fix(provider/kubernetes): ignore cluster-scoped namespace lookup (#2984)

ajordens committed:
fix(core): Handle projects in `allowList` that may not actually exist (#2986)

lwander committed:
fix(provider/kubernetes): fix FCR for cluster-scoped resources w/ namespace (#2985)

lwander committed:
feat(provider/kubernetes): attach lb during deploy (#2988)

jtk54 committed:
fix(provider/gce): Filter instance caching by location. (#2991)

Previously we were including all instances in all locations of a given
server group in each server group written to the cache.

guoyongzhang committed:
feat(artifact/oracle): Added oracle artifact. (#2990)

dibyom committed:
fix(config): Move core config to clouddriver.config package (#2994)

Commit:
fix(provider/gce): Index public COS images. (#2995)

lwander committed:
feat(provider/kubernetes): first pass at disable (#2996)

Does not implement disable yet, but sets up needed scaffolding for
operation, as well ability to build queries for finding pods.

robzienert committed:
chore(core): Add TaskRepositoryTck (#2997)

lwander committed:
fix(provider/kubernetes): validate kind on edit (#2999)

lwander committed:
feat(provider/kubernetes): make checking permissions optional on startup (#3000)

clareliguori committed:
feat(provider/ecs): Use secrets from Secrets Manager for the ECS dock… (#2998)

* feat(provider/ecs): Use secrets from Secrets Manager for the ECS docker image credentials

* groovy fixes

tomaslin committed:
fix(providers/amazon): f1.4xlarge block device mapping (#3002)

tomaslin committed:
fix(provider/titus): fix retry logic when getting duplicate job ids (#3004)

ajordens committed:
perf(core): Attempted move away from fetching _every_ possible Project cluster (#3005)

The current approach to fetching clusters for a Project is quite
inefficient, particular for large applications wherein only a subset
of clusters are included in the project config.

This PR initially fetches cluster summaries and then applies a filter
to determine applicability. The applicable clusters are then fetched in
detail.

tomaslin committed:
feat(provider/titus): don't cache pending scaling policies (#3008)

tomaslin committed:
chore: license headers (#3009)

tomaslin committed:
chore(provider/titus): excludes generated files from javadocs and license header check (#3010)

Commit:
chore(builds): Reformat spinnakerDependenciesVersion (#3012)

spinnakerbot committed:
chore(dependencies): Autobump spinnaker-dependencies (#3013)

Commit:
feat(provider/cf): caching agent backwards compatible to CAPI 2.94.0/3.29.0

tomaslin committed:
feat(provider/titus): streaming titus client (#3006)

ajordens committed:
feat(web): Allow for request queue size to be overridden dynamically (#3019)

```
Most typically, core and maximum pool sizes are set only upon
construction, but they may also be changed dynamically using
setCorePoolSize(int) and setMaximumPoolSize(int).

If the new value is smaller than the current value, excess existing
threads will be terminated when they next become idle. If larger, new
threads will, if needed, be started to execute any queued tasks.
```

^^ from JDK documentation

lwander committed:
fix(artifacts/gcs): fix version resolution (#3020)

maggieneterval committed:
feat(provider/kubernetes): split kubernetes pod logs by container (#3017)

spinnakerbot committed:
chore(dependencies): Autobump spinnaker-dependencies (#3016)

Commit:
feat(provider/kubernetes): k8s resource whitelisting using annotation… (#3014)

ezimanyi committed:
feat(provider/kubernetes): Add Builder to KubernetesImageDescription (#3022)

Some existing code is using an auto-generated constructor for KubernetesImageDescription,
which is fragile as added fields can change the meaning of the constructor. Add a Builder
annotation so this code can use the more robust builder pattern.

spinnakerbot committed:
chore(dependencies): Autobump spinnaker-dependencies (#3025)

spinnakerbot committed:
chore(dependencies): Autobump spinnaker-dependencies (#3026)

Remz-Jay committed:
Merge branch 'master' into feature/2963
